### PR TITLE
Make latest testhost compat with older vstest.console

### PR DESCRIPTION
**Pull request by smadala**
- In absence of endpoint and role arguments to testhost, fallback to defaults.

### CODE REVIEW

File: src/testhost.x86/DefaultEngineInvoker.cs [hunk 1]
```diff
@@ -54,9 +55,21 @@ public void Invoke(IDictionary<string, string> argsDictionary)
             }
 #endif
 
-            // Get port number and initialize communication
+            // vstest.console < 15.5 won't send endpoint, role arguments.
+            // So derive endpoint from port argument and Make connectionRole as Client.
             string endpoint = CommandLineArgumentsHelper.GetStringArgFromDict(argsDictionary, EndpointArgument);
-            ConnectionRole connectionRole = string.Equals(CommandLineArgumentsHelper.GetStringArgFromDict(argsDictionary, RoleArgument), "client", StringComparison.OrdinalIgnoreCase) ? ConnectionRole.Client : ConnectionRole.Host;
+            if (endpoint == string.Empty)
```
Comment (codito) [inline src/testhost.x86/DefaultEngineInvoker.cs]: string.IsNullOrEmpty? Also in line 68

**Review by codito — APPROVED** [review]

**Review by mayankbansal018 — APPROVED** [review]

File: src/testhost.x86/DefaultEngineInvoker.cs (hunk 1 above)
Comment (smadala) [inline src/testhost.x86/DefaultEngineInvoker.cs]: Done.

**Comment by smadala** [comment]
@dotnet-bot test Windows_NT / Debug Build please

**Comment by smadala** [comment]
@dotnet-bot test Windows_NT / Debug Build please.

**Pull request merged**